### PR TITLE
refactor: sort OTC config file contents alphabetically

### PR DESCRIPTION
**Pull request by astencel-sumo**
Refs: https://github.com/SumoLogic/sumologic-kubernetes-collection/issues/2975

### CODE REVIEW

File: deploy/helm/sumologic/conf/events/otelcol/config.yaml [hunk 1]
```diff
@@ -1,51 +1,67 @@
...
+{{- end }}
+
+  ## ref: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.73.0/extension/healthcheckextension
+  health_check: {}
+
+  ## ref: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.73.0/extension/pprofextension
+  pprof: {}
+
 processors:
+  ## ref: https://github.com/open-telemetry/opentelemetry-collector/tree/v0.73.0/processor/batchprocessor
```
[the remark this answers is not in the document]
Comment (astencel-sumo) [reply deploy/helm/sumologic/conf/events/otelcol/config.yaml:30]: Good point, I was going to check this.

**Comment by swiatekm-sumo** [comment]
Do we have some way of checking this in CI? If not, it will slowly diverge from the intended state.

File: deploy/helm/sumologic/conf/events/otelcol/config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (astencel-sumo) [reply deploy/helm/sumologic/conf/events/otelcol/config.yaml:30]: @sumo-drosiek please review https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/2977. Once it's merged, I'll rebase this PR on it and add the necessary replacements.

**Comment by astencel-sumo** [comment]
> Do we have some way of checking this in CI? If not, it will slowly diverge from the intended state.

Do you mean the alphabetic sorting? No, we don't have a way to check this in CI. We cannot use any YAML manipulation like yq, because the files contain Go template strings. I don't see a way to do this at the moment.

**Comment by sumo-drosiek** [comment]
> > Do we have some way of checking this in CI? If not, it will slowly diverge from the intended state.
> 
> Do you mean the alphabetic sorting? No, we don't have a way to check this in CI. We cannot use any YAML manipulation like yq, because the files contain Go template strings. I don't see a way to do this at the moment.

Can we use `helm template` and `yq` for that?

File: deploy/helm/sumologic/conf/events/otelcol/config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (astencel-sumo) [reply deploy/helm/sumologic/conf/events/otelcol/config.yaml:30]: Added.

**Comment by swiatekm-sumo** [comment]
To be clear, I'm good merging this as-is, but I also think we should create an issue to check this in CI if it's feasible.

**Pull request merged**